### ISSUE TITLE
AxiosError: Network Error in react native when sending a POST require with formData

### ISSUE THREAD

**Issue by PAULbigBA** [axios/axios#6741]
### Describe the issue

When I want to send a POST requirement with formData, I include one Image(File type) and one JSON text (blob type) in that formData, it will raise an error: " AxiosError: Network Error " in react native (mobile platform). It is working fine on the web platform.
I have tried many ways to solve this and try to debug it.
1. I manually or automatically changed the content type in the header to 'multipart/form-data'.
2. Rollback to 1.7.2 from 1.7.7 since 1.7.2 is the previous version I was able to send successfully before it was on the mobile platform(only test on Android)
3. Check react native support File and Blob.(I send a formdata with empty content, it still raise same error).
4. Already add android:usesCleartextTraffic="true".
5. Check isFormData function in the utils.js file by sending an empty content #3.
6. check whether I include HTTP://
![1734454832258](https://github.com/user-attachments/assets/afa7b4ea-4903-4a81-b513-4bf4f0a35adc)

### Example Code

```js
const formData = new FormData();
        if (preparedFile instanceof File) {
          formData.append('file', preparedFile);
        } else {
          formData.append('file', {
            uri: preparedFile.uri,
            type: preparedFile.type,
            name: preparedFile.name,
          } as unknown as File);
        }

        const upload = {
          KEY: 'UPLOAD'
        };
        const json = JSON.stringify(upload );
        const blob = new Blob([json], {
          type: 'application/json',
        });

        formData.append('upload', blob);

        await createUpload(formData);
```

### Axios Version

1.7.7

### Adapter Version

xhr/http/fetch

### Browser

Microsoft Edge

### Browser Version

131.0.2903.99

### Node.js Version

20.11.1

### OS

Android 11 API 30

### Additional Library Versions

```bash
"react": "18.3.1",
"react-native": "0.76.2",
"expo": "52.0.7",
```

**Comment by John251314** [comment]
yeah, I face the same issue! It happend only on android platform, and mybe success or failed. 
I got the error, and found request.status is 0 and request._response is "Stream closed", I dont know what happend!
<img width="672" alt="image" src="https://github.com/user-attachments/assets/abf5acb0-75de-4f72-8d29-560080c56453" />

I use react-native-crop-image-pick take photo and use axios upload to server，if someone has the solution，please tell me. 
 3Q！

**Comment by suhaotian** [comment]
Can you try with native fetch, is it work?

**Comment by MedtheVorg** [comment]
use native  fetch it works fine.
example code 

<img width="540" height="382" alt="Image" src="https://github.com/user-attachments/assets/b573838f-921b-4c02-a3f9-9101b3596293" />

this works because  react-native has a native implementation for the fetch function and when we set the file in this way

<img width="497" height="143" alt="Image" src="https://github.com/user-attachments/assets/44fc3a28-b19e-431e-a1aa-df88e0da03e8" />

it recognizes this pattern and handles creating proper multipart data request and attaches the file data using its uri.


<img width="1069" height="155" alt="Image" src="https://github.com/user-attachments/assets/c2b4b7b8-5dfe-4704-b48c-7c8420a2a6fb" />

**Comment by jasonsaayman** [comment]
Duplicates #6968
